integration/simulation/simulation_test.go

```diff
@@ -97,21 +97,23 @@ func checkBlockchainValidity(t *testing.T, txManager *TransactionManager, networ
 // validateL1L2Stats validates blockchain wide properties between L1 and the L2
 func validateL1L2Stats(t *testing.T, node *host.Node, stats *Stats) {
 	l1Height := obscurocommon.L1GenesisHeight
-	for header := node.DB().GetCurrentBlockHead(); header != nil && header.ID != obscurocommon.GenesisHash; header = node.DB().GetBlockHeader(header.Parent) {
+	for header := node.DB().GetCurrentBlockHead(); header != nil; header = node.DB().GetBlockHeader(header.Parent) {
 		l1Height++
 	}
-	l2Height := obscurocommon.L2GenesisHeight
-	for header := node.DB().GetCurrentRollupHead(); header != nil && header.ID != obscurocommon.GenesisHash; header = node.DB().GetRollupHeader(header.Parent) {
+	//  If GetCurrentBlockHead is block height 10 and it stops at block 0 it counts 11. 11 - 1 = 10
```

Comment:
I don't quite understand the reasoning behind this

Reply:
Yeah that was bugging me too! :) 

We're double counting a block. Between block 0 and block 10 there are 11 blocks, but the final height is 10.

The "correct" way to do this would be `l2height = obscurocommon.L2GenesisHeight + counter` but we need to remove the first block.
The L2GenesisHeight is already a block so we need to not count it.
```
height = genesisHeight + countAllBlocks - Oneblock
10 = 0 + 11 // invalid: it counted block 0 as a block
10 = 0 + 11 -1 // valid: we removed the first block count
110 = 100 + 11 - 1 // valid

Reply:
ah, I think it's because that for loop travels all the way back to the virtual block before the genesis which is null. And that block obviously doesn't exist so it has to be removed from the height